cmd/jaeger/internal/integration/badger_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"github.com/jaegertracing/jaeger/pkg/testutils"
+	"github.com/jaegertracing/jaeger/plugin/storage/badger"
+	"github.com/jaegertracing/jaeger/plugin/storage/integration"
+)
+
+type BadgerStorageIntegration struct {
+	E2EStorageIntegration
+	logger  *zap.Logger
+	factory *badger.Factory
+}
+
+func (s *BadgerStorageIntegration) initialize(t *testing.T) {
+	s.factory = badger.NewFactory()
+
+	err := s.factory.Initialize(metrics.NullFactory, zap.NewNop())
+	require.NoError(t, err)
+
+	s.SpanWriter, err = s.factory.CreateSpanWriter()
+	require.NoError(t, err)
+
+	s.SpanReader, err = s.factory.CreateSpanReader()
+	require.NoError(t, err)
+
+	s.SamplingStore, err = s.factory.CreateSamplingStore(0)
+	require.NoError(t, err)
```

Comment:
I don't think this is correct. These lines of code replace the initialized SpanWriter and SpanReader from e2eInitialize() function, which writes span data to the OTLP receiver and reads from jaeger_query. But these read/write from badger storage directly and are not the goal of integration tests in `cmd/jaeger/internal/integration`.

Reply:
Overall my preference would be to have the individual test be only responsible for constructing a factory and passing it over to the overall framework. Right now we're still repeating a lot of boilerplate code in each storage test.

Reply:
And in case of e2e tests even the factory is not necessary, only implementing the cleanup() function is needed to delete data between tests (incidentally, if we were to better support tenancy, even deletion of the data would not be necessary).

Reply:
>I don't think this is correct. These lines of code replace the initialized SpanWriter and SpanReader from e2eInitialize() function, which writes span data to the OTLP receiver and reads from jaeger_query. But these read/write from badger storage directly and are not the goal of integration tests in cmd/jaeger/internal/integration.

oops sorry, i'll fix it asap